csrc/scheduler/reduction.cpp

```diff
@@ -507,29 +507,270 @@ std::shared_ptr<ReductionParams> innerReductionHeuristic(
   return rparams;
 }
 
+struct outerReduHeuristicParas {
+  outerReduHeuristicParas(
+      int64_t total_iteration_numel,
+      int64_t total_reduction_numel)
+      : total_iteration_numel(total_iteration_numel),
+        total_reduction_numel(total_reduction_numel) {}
+  // iteration dim paras
+  // iteration elements = iter_unroll * bdimx * gdimx
+  int64_t iter_unroll_factor = 1;
+  int64_t bdimx = 1;
+  int64_t gidim = 1;
```

Comment:
is `gidim` supposed to be `gdimx`? code comment seems to be inconsistent with the variable name here.